### ISSUE TITLE
Compass teleport triggers unexpected 

### ISSUE THREAD

**Issue by Schneidertm** [EngineHub/WorldEdit#1582]
**Versions**
WorldEdit version: Bukkit-Official(7.2.0+a51fa43)
Platform version: git-Paper-271 (1.16.4)

**Describe the bug**
Teleport commands (/jumpto and /thru ) which are bound to the compass does trigger while equipped when dropping items out of the inventory

**To Reproduce**
1. Permissions to use the commands /jumpto and /thru & commands are bound to the compass item
2. Hold the compass in your right hand
3. Open the inventory
4. Drop a random item with the shortcut key q 

**Expected behavior**
The compass only teleports when the user is aware and not blocked by interfaces such as the inventory.

**Screenshots**
https://www.youtube.com/watch?v=948aBTFVJwI

**Comment by octylFractal** [comment]
I would report this to Spigot or Paper, as they are the ones who provide the left click event. Likely they need to not send the event if a GUI is open.

**Comment by nathanfranke** [comment]
Possibly related to https://github.com/PaperMC/Paper/issues/3289, but it was marked as a wontfix
